lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Shared.hs

```diff
@@ -597,3 +601,18 @@ liftDelegationAddress ix dTemplate (KeyFingerprint fingerprint) =
     delegationCredential = DelegationFromScript . toScriptHash
     dScript =
         replaceCosignersWithVerKeys CA.Stake dTemplate ix
+
+toSharedWalletId
+    :: (WalletKey k, k ~ SharedKey)
+    => k 'AccountK XPub
+    -> ScriptTemplate
+    -> Maybe ScriptTemplate
+    -> Digest Blake2b_160
+toSharedWalletId accXPub pTemplate dTemplateM =
+    hash $
+    (unXPub . getRawKey $ accXPub) <>
+    toByteString pTemplate <>
+    maybe mempty toByteString dTemplateM
+  where
+    toByteString (ScriptTemplate _ script) =
```

Comment:
Commented on `cardano-addresses`: The use of `show` makes it more difficult for non-Haskell code to construct wallet IDs. I propose to export

```hs
serializeScriptTemplate :: Script Cosigner -> ByteString
```

from `cardano-addresses` and then use it here.

… Later, after version bumps.

Reply:
reused bits from cardano-addresses